### DISCUSSION TITLE
Adding React custom detail components into existing Admin UI Pages

### DISCUSSION THREAD

**Discussion by charleshopfner** [vendurehq/vendure#2082] [Q&A] [answered]
Hello,

We are working on a POC to see if Vendure could answer the needs of our client.
One of the things we have to try is to implement custom React component into an existing admin-UI page.
Let's say we want to add a component to the customer detail page, how could we do this ? Is it possible ?

I think I saw on another topic that seemed related, that it would be difficult to do, and would not be as good as if we used Angular for this purpose. If yes, is it still the case ?

Thanks in advance for your answer.

**Comment by michaelbromley** [comment, marked as answer]
Hi,

Currently it is not possible to use a React component for extending an existing detail page such as the customer detail page. You'd need to use Angular, as described in the [Custom Detail Components docs](https://www.vendure.io/docs/plugins/extending-the-admin-ui/custom-detail-components/). React _can_ be used to create [entirely new routes](https://www.vendure.io/docs/plugins/extending-the-admin-ui/using-other-frameworks/) though, just not embedded inside existing pages.

In the future I would like to allow React to be used in all these places, but it's not immediately planned.
